index.js

```diff
@@ -160,7 +169,76 @@ class TuyaDevice extends EventEmitter {
           resolve(data.dps['1']);
         }
       })
-        .catch(reject);
+          .catch(reject);
+    });
+  }
+
+  /**
+   * Refresh a device's current status.
+   * Defaults to returning only the value of the first DPS index.
+   * @param {Object} [options]
+   * @param {Boolean} [options.schema]
+   * true to return entire list of properties from device
+   * @param {Array.<Number>} [options.dps=[4,5,6,18,19,20]]
+   * DPS index to return
+   * @example
+   * // get first, default property from device
+   * tuya.refresh().then(status => console.log(status))
+   * @example
+   * // get second property from device
+   * tuya.refresh({dps: [4,5,9,18,19,20]}).then(status => console.log(status))
+   * @example
+   * // get all available data from device
+   * tuya.refresh({schema: true}).then(data => console.log(data))
+   * @returns {Promise<Boolean|Object>}
+   * returns boolean if single property is requested, otherwise returns object of results
+   */
+  refresh(options = {}) {
+    const payload = {
+      gwId: this.device.gwID,
+      devId: this.device.id,
+      t: Math.round(new Date().getTime() / 1000).toString(),
+      dpId: this._dpRefreshIds,
```

Comment:
This is consuming the instance options instead of the function options, if that's the intended behavior the above documentation needs to be updated.

Reply:
Nope, it was mistake. Now it should be okay.